Documentation/README.md

```diff
@@ -57,3 +57,130 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
 cd <path-to-extraction-root>
 ./build.sh
 ```
+## Building a custom .NET Runtime
+
+You can also build a .NET Core Runtime with custom changes in order to test locally the produced runtime with the affected changes.
+
+The .NET Core Runtime consists of 4 repos, core-setup at the top of the stack (which produces the runtime), standard, corefx (BCL repo), coreclr (Runtime repo). 
+
+This means that in order to get a runtime you'll need to build the source-build repo with the `/p:RootRepo=core-setup` parameter, this will automatically build coreclr, standard and corefx. Source-Build is composed by git submodules, so the first thing you need to run before doing anything is:
+
+```console
+git submodule --init --recursive
+```
+
+This will initialize the submodules and clone the repos under the src folder if they haven't been initialized. Note that this will point the submodule to the branch and commit that the submodule is pointing to. However you can go to each repo indivually, add custom remotes and checkout certain branches containing your changes.
+
+Let's say that you want to produce a runtime that has a new API that it's implementation lives in coreclr and you want to create a .NET Core console app that uses it to test it without waiting for a new SDK or Runtime to be produced by official builds. Imagine we will add an API that just throws in Hashtable. 
```

Comment:
I think adding a `###` header before this paragraph would be nice--`Walkthrough` or `Example .NET Runtime build from source` or something. It seems like the above info is enough to perform the build for someone who already understands submodules and has read the rest of the doc above this--they they can skip ahead to `How to use the produced Runtime`.